sway/commands/input/scroll_factor.c

```diff
@@ -0,0 +1,26 @@
+#include <stdlib.h>
+#include <string.h>
+#include "sway/config.h"
+#include "sway/commands.h"
+#include "sway/input/input-manager.h"
+
+struct cmd_results *input_cmd_scroll_factor(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "scroll_factor", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+	struct input_config *ic = config->handler_context.input_config;
+	if (!ic) {
+		return cmd_results_new(CMD_FAILURE,
+			"scroll_factor", "No input device defined.");
+	}
+
+	float scroll_factor = atof(argv[0]);
+	if (scroll_factor < 0) {
+		return cmd_results_new(CMD_INVALID, "scroll_factor",
+			"Input out of range [0, inf)");
```

Comment:
The brackets aren't matching. Although I'd prefer wording such as "Scroll factor cannot be negative".

Reply:
I think it's deliberate, it's a math notation. But yeah natural language seems more appropriate here.